pkg/operator/ceph/cluster/osd/spec.go

```diff
@@ -74,12 +75,23 @@ set -o nounset # fail if variables are unset
 set -o xtrace
 
 OSD_ID="$ROOK_OSD_ID"
+CEPH_FSID=%s
 OSD_UUID=%s
 OSD_STORE_FLAG="%s"
 OSD_DATA_DIR=/var/lib/ceph/osd/ceph-"$OSD_ID"
 CV_MODE=%s
 DEVICE="$%s"
 
+# "ceph.conf" must have the "fsid" global configuration after the following Ceph's PR is merged.
+# https://github.com/ceph/ceph/commit/25655e5a8829e001adf467511a6bde8142b0a575
+cp --no-preserve=mode /etc/temp-ceph/ceph.conf /etc/ceph/ceph.conf
+if grep -q "^\[global]$" /etc/ceph/ceph.conf ; then
```

Comment:
I'm trying to better understand, the only `ceph.conf` we had was coming out of the config override CM, right?

How about the following python?

```python
python3 -c "
import configparser
config = configparser.ConfigParser()
config.read('ceph.conf')


if not config.has_option('global','fsid'):
    config['global']['fsid'] = 'bar'

with open('ceph.conf', 'w') as configfile:
    config.write(configfile)
"
```

It looks safer than the bash snippet and we use python in other places too.
Let me know what you think.

Reply:
> I'm trying to better understand, the only ceph.conf we had was coming out of the config override CM, right?

That's right.

> How about the following python?

It looks better than what I wrote. I'll update this PR later.